sql/parser/eval.go

```diff
@@ -650,25 +650,28 @@ func evalUnaryExpr(expr *UnaryExpr) (Datum, error) {
 
 func evalFuncExpr(expr *FuncExpr) (Datum, error) {
 	name := strings.ToLower(expr.Name.String())
-	b, ok := builtins[name]
-	if !ok {
-		return DNull, fmt.Errorf("%s: unknown function", expr.Name)
-	}
-	if b.nArgs != -1 && b.nArgs != len(expr.Exprs) {
-		return DNull, fmt.Errorf("%s: incorrect number of arguments: %d vs %d",
-			expr.Name, b.nArgs, len(expr.Exprs))
-	}
-
 	args := make(DTuple, 0, len(expr.Exprs))
-	for _, e := range expr.Exprs {
+	var types [5]string
+	for i, e := range expr.Exprs {
 		arg, err := EvalExpr(e)
 		if err != nil {
 			return DNull, err
 		}
 		args = append(args, arg)
+		types[i] = arg.Type()
```

Comment:
How about using string as the key, and the string is computed from the method name and args types and optional args,  like ```func (sig builtin) String() string```. 
This could also eliminate the need to regiser the function twice for functions with optional arguments.